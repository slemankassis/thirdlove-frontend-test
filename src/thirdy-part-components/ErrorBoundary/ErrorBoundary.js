import React from 'react';
import PropTypes from 'prop-types';

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  componentDidCatch() {
    this.setState({ hasError: true });
  }

  render() {
    if (this.state.hasError) {
      return <div className="error-boundary">{this.props.errorMessage}</div>;
    }
    return this.props.children;
  }
}

ErrorBoundary.propTypes = {
  errorMessage: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element),
  ]).isRequired,
};

ErrorBoundary.defaultProps = {
  errorMessage: 'There was an error loading this component.',
};

export default ErrorBoundary;
