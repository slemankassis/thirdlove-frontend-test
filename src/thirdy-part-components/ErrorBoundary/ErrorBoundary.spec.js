import React from 'react';
import ErrorBoundary from './ErrorBoundary';

describe('ErrorBoundary', () => {
  it('renders snapshot with basicProps', () => {
    const basicProps = {
      selected: {
        value: '__VALUE__',
        label: '__LABEL__',
      },
      options: ['__OPTION_1__', '__OPTION_2__'],
      onChange: jest.fn(),
    };

    const component = shallow(<ErrorBoundary {...basicProps} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('renders snapshot with fullProps', () => {
    const fullProps = {
      selected: {
        value: '__VALUE__',
        label: '__LABEL__',
      },
      options: ['__OPTION_1__', '__OPTION_2__'],
      onChange: jest.fn(),
      errorMessage: '__ERROR_MESSAGE__',
      children: React.createElement('div'),
    };

    const component = shallow(<ErrorBoundary {...fullProps} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });
});
