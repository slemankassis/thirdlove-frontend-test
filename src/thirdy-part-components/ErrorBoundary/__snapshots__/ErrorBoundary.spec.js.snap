// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary renders snapshot with basicProps 1`] = `""`;

exports[`ErrorBoundary renders snapshot with fullProps 1`] = `"<div />"`;
